performance/src/performance/timer.cpp

```diff
@@ -39,8 +39,8 @@ timer::~timer() {
     const std::chrono::nanoseconds totalTime = end - m_start;
     const auto pos =
         std::find_if(m_timing_info.data.begin(), m_timing_info.data.end(),
-                     [&m_name = m_name](const timing_info_pair& element) {
-                         return element.first == m_name;
+                     [&name = m_name](const timing_info_pair& element) {
+                         return element.first == name;
```

Comment:
```suggestion
                     [&m_name](const timing_info_pair& element) {
                         return element.first == m_name;
```

Reply:
Also should be `std::find` (see below), but no need to fix that in this PR.

Reply:
I had to try this out to remind myself why the code was written in such a weird way to begin with. Unfortunately capture by reference does not like to capture member variables all that much.

```
/data/ssd-1tb/projects/traccc/traccc/performance/src/performance/timer.cpp:42:24: error: 'm_name' in capture list does not name a variable
                     [&m_name](const timing_info_pair& element) {
                       ^
/data/ssd-1tb/projects/traccc/traccc/performance/src/performance/timer.cpp:43:50: error: 'this' cannot be implicitly captured in this context
                         return element.first == m_name;
                                                 ^
/data/ssd-1tb/projects/traccc/traccc/performance/src/performance/timer.cpp:42:30: note: explicitly capture 'this'
                     [&m_name](const timing_info_pair& element) {
                             ^
                             this
```

If I remember correctly, @Xola101 was capturing `this` in the lambda at first. Which I myself didn't like. That's how we ended up with `[&m_name = m_name]`. :thinking:

All in all, I'd rather leave the code as I have it now...